File: tests/html/rtc_test.dart
// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

library RealTimeCommunicationTest;

import 'package:unittest/unittest.dart';
import 'package:unittest/html_individual_config.dart';
import 'dart:html';

main() {
  useHtmlIndividualConfiguration();

  group('supported', () {
    test('supported', () {
      expect(RtcPeerConnection.supported, true);
    });
  });

  group('functionality', () {
    // More thorough testing of this API requires the user to
    // explicitly click "allow this site to access my camera and/or microphone."
    // or particularly allow that site to always have those permission on each
    // computer the test is run. For more through tests, see
    // interactive_test.dart.
    if (RtcPeerConnection.supported) {
      test('peer connection', () {
        var pc = new RtcPeerConnection({
          'iceServers': [
            {'url': 'stun:216.93.246.18:3478'}
          ]
        });
        expect(pc is RtcPeerConnection, isTrue);
        // TODO(efortuna): Uncomment this test when RTCPeerConnection correctly
        // implements EventListener in Firefox (works correctly in nightly, so
        // it's coming!).
        //pc.onIceCandidate.listen((candidate) {});
      });

      test('ice candidate', () {
        var candidate =
            new RtcIceCandidate({'sdpMLineIndex': 1, 'candidate': 'hello'});
        expect(candidate is RtcIceCandidate, isTrue);
      });

      test('session description', () {
        var description =
            new RtcSessionDescription({'sdp': 'foo', 'type': 'offer'});
        expect(description is RtcSessionDescription, isTrue);
      });
    }
  });
}

File: pkg/dev_compiler/test/codegen_expected/lib/html/rtc_test.js
dart_library.library('lib/html/rtc_test', null, /* Imports */[
  'dart_sdk',
  'unittest'
], function load__rtc_test(exports, dart_sdk, unittest) {
  'use strict';
  const core = dart_sdk.core;
  const html = dart_sdk.html;
  const _interceptors = dart_sdk._interceptors;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const html_individual_config = unittest.html_individual_config;
  const unittest$ = unittest.unittest;
  const src__matcher__expect = unittest.src__matcher__expect;
  const src__matcher__core_matchers = unittest.src__matcher__core_matchers;
  const rtc_test = Object.create(null);
  let MapOfString$String = () => (MapOfString$String = dart.constFn(core.Map$(core.String, core.String)))();
  let JSArrayOfMapOfString$String = () => (JSArrayOfMapOfString$String = dart.constFn(_interceptors.JSArray$(MapOfString$String())))();
  let ListOfMapOfString$String = () => (ListOfMapOfString$String = dart.constFn(core.List$(MapOfString$String())))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  let VoidTovoid = () => (VoidTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [])))();
  rtc_test.main = function() {
    html_individual_config.useHtmlIndividualConfiguration();
    unittest$.group('supported', dart.fn(() => {
      unittest$.test('supported', dart.fn(() => {
        src__matcher__expect.expect(html.RtcPeerConnection[dartx.supported], true);
      }, VoidTodynamic()));
    }, VoidTovoid()));
    unittest$.group('functionality', dart.fn(() => {
      if (dart.test(html.RtcPeerConnection[dartx.supported])) {
        unittest$.test('peer connection', dart.fn(() => {
          let pc = html.RtcPeerConnection.new(dart.map({iceServers: JSArrayOfMapOfString$String().of([dart.map({url: 'stun:216.93.246.18:3478'}, core.String, core.String)])}, core.String, ListOfMapOfString$String()));
          src__matcher__expect.expect(html.RtcPeerConnection.is(pc), src__matcher__core_matchers.isTrue);
        }, VoidTodynamic()));
        unittest$.test('ice candidate', dart.fn(() => {
          let candidate = html.RtcIceCandidate.new(dart.map({sdpMLineIndex: 1, candidate: 'hello'}, core.String, core.Object));
          src__matcher__expect.expect(html.RtcIceCandidate.is(candidate), src__matcher__core_matchers.isTrue);
        }, VoidTodynamic()));
        unittest$.test('session description', dart.fn(() => {
          let description = html.RtcSessionDescription.new(dart.map({sdp: 'foo', type: 'offer'}, core.String, core.String));
          src__matcher__expect.expect(html.RtcSessionDescription.is(description), src__matcher__core_matchers.isTrue);
        }, VoidTodynamic()));
      }
    }, VoidTovoid()));
  };
  dart.fn(rtc_test.main, VoidTodynamic());
  // Exports:
  exports.rtc_test = rtc_test;
});
